### ISSUE TITLE
Styling a element with several CSS classes

### ISSUE THREAD

**Issue by existme** [arve0/markdown-it-attrs#75]
Hi,
I am trying to use this in `tui.editor` within a Gollum installation, I have already written a render layer for Gollum which mimics `markdown-it-attrs` behaviors. Something that makes me wonder is some inconsistencies when having several classes on an element.

Example input:
```md
1.this is the markdown I'm trying to parse {.note .green}
2.this is the markdown I'm trying to parse {.note.green}
```
Current output:
```html
<p class="note green">1.this is the markdown I'm trying to parse</p>
<p>2.this is the markdown I'm trying to parse</p>
```

Expected output:
```html
<p class="note green">1.this is the markdown I'm trying to parse</p>
<p class="note green">2.this is the markdown I'm trying to parse</p>
```
Won't you agree that both should render the same way? I mean why not allowing the shorthand `.warning.green` instead of `.warning .green`? and why it completely eats `.note.green`?

Thanks
/Reza

**Comment by arve0** [comment]
Hi, thank you for reporting 🙂
Turns out, `.` is [allowed in class names](https://stackoverflow.com/questions/448981/which-characters-are-valid-in-css-class-names-selectors#answer-6732899).

You can try it out with this HTML:

```html
<html lang="en">
<head>
    <style>
    .a\.b {
        color: yellow;
    }
    </style>
</head>
<body>
    <p class="a.b">hello</p>
</body>
</html>
```

Regarding the output, it is actually

```
<p class="note green">1.this is the markdown I'm trying to parse</p>
<p css-module="notegreen">2.this is the markdown I'm trying to parse</p>
```

which seems to be a bug. I'll fix it.

**Comment by existme** [comment]
Thanks for the really quick feedback. Regarding the output, If you say so, then this is something that `tui.editor` does on top and eats the `css-module` sorry for the confusion. But don't you think that it's better to render `{.note.green}` -> `class="note green"` and `{.note\.green}` -> `class="note.green"`.

**Comment by arve0** [comment]
> But don't you think that it's better to render {.note.green} -> class="note green" and {.note\.green} -> class="note.green".

I understand why you ask, as `.a.b` in CSS means "class a and b on same element`. Though I see at least three reasons not parse it that way:

1. Pandoc (that markdown-it-attrs mimics) renders `{.with.dot}` as `class="with.dot"`.
2. Markdown is nearer to HTML (`<p class="separate with spaces">`) then to CSS (`.multiple.classes.on.same.element { ... }`).
3. The extra space, `{.note .green}`, is an explicit communication of "this is two things". Explicit > implicit to me at least.

**Comment by existme** [comment]
Fair enough :+1: Thanks for the quick feedback, patch and the explanation :)

**Comment by arve0** [comment]
No problem, fix in `v2.3.2`.
